game-core/src/main/java/games/strategy/triplea/ui/MapPanel.java

```diff
@@ -677,15 +677,15 @@ public void paint(final Graphics g) {
     final Rectangle2D.Double mainBounds =
         new Rectangle2D.Double(x, y, getScaledWidth(), getScaledHeight());
     drawTiles(g2d, images, data, mainBounds, undrawnTiles);
-    if (routeDescription != null && mouseShadowImage != null && routeDescription.getEnd() != null) {
-      final AffineTransform t = new AffineTransform();
-      t.translate(
-          normalizeX(routeDescription.getEnd().getX() - getXOffset()),
-          normalizeY(routeDescription.getEnd().getY() - getYOffset()));
-      t.translate(mouseShadowImage.getWidth() / -2.0, mouseShadowImage.getHeight() / -2.0);
-      g2d.drawImage(mouseShadowImage, t, this);
-    }
     if (routeDescription != null) {
+      if (mouseShadowImage != null && routeDescription.getEnd() != null) {
+        final AffineTransform t = new AffineTransform();
+        t.translate(
+            normalizeX(routeDescription.getEnd().getX() - getXOffset()),
+            normalizeY(routeDescription.getEnd().getY() - getYOffset()));
+        t.translate(mouseShadowImage.getWidth() / -2.0, mouseShadowImage.getHeight() / -2.0);
+        g2d.drawImage(mouseShadowImage, t, this);
+      }
```

Comment:
Not completely following this change. Can you add some context?

Reply:
This is just a semantic change.
Previously both if cases checked if RouteDescription is non null, now there's a single if check and a nested if statement.

It's just a random change I had lying around locally.